### PR TITLE
[WIP] Fix a bug in Drools-based Constraint Streams

### PR DESCRIPTION
**Pull request by triceo**
This doesn't actually fix anything yet. :-)

### CODE REVIEW

File: optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/BavetConstraintSession.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ public void update(Object fact) {
 
     @Override
     public void retract(Object fact) {
-        List<BavetFromUniTuple<Object>> tupleList = fromTupleListMap.get(fact);
+        List<BavetFromUniTuple<Object>> tupleList = fromTupleListMap.remove(fact);
```
Comment (triceo) [inline optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/BavetConstraintSession.java]: I think I found (and fixed) a bug in Bavet.

Comment (ge0ffrey) [reply optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/BavetConstraintSession.java]: Agreed. nice catch!

Can we have a unit test? :) (other PR is fine)

File: optaplanner-core/src/test/java/org/optaplanner/core/api/score/stream/bi/BiConstraintStreamTest.java [hunk 2]
```diff
@@ -53,6 +54,11 @@ public BiConstraintStreamTest(boolean constraintMatchEnabled, ConstraintStreamIm
     // Filter
     // ************************************************************************
 
+    @Before
+    public void assumeBavet() {
+        super.assumeBavet(); // skip tests if we're not using Bavet
```
Comment (ge0ffrey) [inline optaplanner-core/src/test/java/org/optaplanner/core/api/score/stream/bi/BiConstraintStreamTest.java]: code style: the s of "skip" with capital.

File: optaplanner-core/src/test/java/org/optaplanner/core/api/score/stream/uni/MarioConstraintStreamTest.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    @Before
+    public void skipUnnecessaryTests() {
+        Assume.assumeTrue(constraintStreamImplType == ConstraintStreamImplType.DROOLS);
+        Assume.assumeTrue(constraintMatchEnabled);
+    }
+
+    @Test
+    public void usingPlainDRL() {
```
Comment (triceo) [inline optaplanner-core/src/test/java/org/optaplanner/core/api/score/stream/uni/MarioConstraintStreamTest.java]: @ge0ffrey Run this. It fails. Blows my mind.

**Comment by triceo** [comment]
Replaced by #566.

**Pull request closed without merging**